compiler/rustc_resolve/src/imports.rs

```diff
@@ -1261,10 +1261,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         *module.globs.borrow_mut() = Vec::new();
 
         if let Some(def_id) = module.opt_def_id() {
+            let mut non_reexports = Vec::new();
             let mut reexports = Vec::new();
 
             module.for_each_child(self, |this, ident, _, binding| {
-                if let Some(res) = this.is_reexport(binding) {
+                let res = binding.res().expect_non_local();
+                if !binding.is_import() {
+                    non_reexports.push(res.def_id().expect_local());
+                } else if res != def::Res::Err && !binding.is_ambiguity() {
```

Comment:
For `Res::Err` and ambiguities, should this closure just be a no-op?

Reply:
Yes, those are effectively imports that import nothing, at least from other crates' points of view.
(Although for ambiguities it's an open question - https://github.com/rust-lang/rust/issues/36837.)